### PR TITLE
SC-9023 - Homeworks Dashboard

### PR DESCRIPTION
**Pull request by dominikbruechner**
# Short Description

- Shows open homeworks in a separate beta/dashboard route

## Checklist before merging

- [ ] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [ ] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/components/organisms/HomeworksList.vue [hunk 1]
```diff
@@ -0,0 +1,52 @@
+<template>
+	<v-list subheader two-line>
+		<v-subheader>{{ $t("pages.homeworks.subtitleOpen") }}</v-subheader>
+		<template v-for="(homework, index) of homeworks">
```
[the remark this answers is not in the document]
Comment (dominikbruechner) [reply src/components/organisms/HomeworksList.vue:4]: Empty States we are implementing as part of the next sprint. 

A Loading State is a very good point to keep in mind, also showing an error if the return of a result fails.

[the remark this answers is not in the document]
Comment (CeEv) [reply src/components/organisms/HomeworksList.vue:4]: out of sprint scope -> put to miro

File: src/components/organisms/HomeworksList.vue [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+			<v-list-item :key="homework._id" :href="'/homework/' + homework._id">
```
[the remark this answers is not in the document]
Comment (CeEv) [reply src/components/organisms/HomeworksList.vue:5]: out of sprint scope -> put to miro

File: src/store/homeworks.unit.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+				storeModule.actions.$axios = {
+					$get: async (url, params) => {
+						receivedRequests.push({ url, params });
+					},
+				};
+
+				await storeModule.actions.getHomeworksDashboard(ctxMock);
+				expect(spyCommit.mock.calls).toHaveLength(1);
+				expect(spyCommit.mock.calls[0][0]).toBe("set");
+				expect(receivedRequests[0].url).toStrictEqual("/v3/task/dashboard/");
```
[the remark this answers is not in the document]
Comment (CeEv) [reply src/store/homeworks.unit.js:20]: todo: think about it :D

File: src/components/organisms/HomeworksList.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (dominikbruechner) [reply src/components/organisms/HomeworksList.vue]: resolved

File: src/store/homeworks.js [hunk 4]
```diff
@@ -1,8 +1,42 @@
+import { merge } from "lodash";
 import serviceTemplate from "@utils/service-template";
 const base = serviceTemplate("homework");
```
Comment (pkleybolte) [inline src/store/homeworks.js:3]: I'm not a big fan of the `serviceTemplate`. I prefer actions designed for a specific task over general CRUD operations.

You are no using any of the provided actions. If you don't need it, don't use it.

File: src/components/organisms/HomeworksList.vue [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+export default {
+	components: {},
+	data() {
+		return {
+			fromNow,
+			taskImage: taskImage,
+		};
+	},
+	computed: {
+		...mapGetters("homeworks", {
```
Comment (pkleybolte) [inline src/components/organisms/HomeworksList.vue:47]: My personal opinion is, that the component that triggers the loading of data should also get it from the store and pass the data on. So far this component is not reused so it does not matter that much.

If it is reused:

- Should all instances of the list show the same items based on the store? -> access store here
- Should different instances of the list show different items based on what search params the parent set? ->access store in parent

**Review by pkleybolte — CHANGES_REQUESTED** [review]
My comments are minor issues. 
But since most tests are still todo I don't approve yet.

File: src/components/organisms/HomeworksList.vue (hunk 5 above)
Comment (dominikbruechner) [reply src/components/organisms/HomeworksList.vue:47]: We will choose Should different instances of the list show different items based on what search params the parent set? ->access store in parent , as we probably need to filter data depending on where the widget will be used

Comment (dominikbruechner) [reply src/components/organisms/HomeworksList.vue:47]: generally agreed

File: src/store/homeworks.js (hunk 4 above)
Comment (dominikbruechner) [reply src/store/homeworks.js:3]: as we discussed: we are just using the state template and the commit('set') functionality, which is well implemented in the service-template, so we stick for now with the service-template

File: src/pages/beta/dashboard/index.vue [hunk 6]
```diff
@@ -0,0 +1,31 @@
+<template>
```
[the remark this answers is not in the document]
Comment (dominikbruechner) [reply src/pages/beta/dashboard/index.vue:1]: yes

File: src/pages/beta/dashboard/index.vue [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+	</v-container>
+</template>
+
+<script>
+import HomeworksList from "@components/organisms/HomeworksList";
+
+export default {
+	components: { HomeworksList },
+	layout: "defaultVuetify",
+	created() {
```
[the remark this answers is not in the document]
Comment (dominikbruechner) [reply src/pages/beta/dashboard/index.vue]: done by @martynakrysinska

File: src/components/organisms/HomeworksList.unit.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+	let vuetify;
+
+	beforeEach(() => {
+		vuetify = new Vuetify();
+	});
+
+	it(...isValidComponent(HomeworksList));
+
+	it("Should render complete homework items list", async () => {
+		const wrapper = mount(HomeworksList, {
```
[the remark this answers is not in the document]
Comment (dominikbruechner) [reply src/components/organisms/HomeworksList.unit.js:15]: resolved by @martynakrysinska

File: src/components/organisms/HomeworksList.unit.js [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+	let vuetify;
+
+	beforeEach(() => {
+		vuetify = new Vuetify();
+	});
+
+	it(...isValidComponent(HomeworksList));
+
+	it("Should render complete homework items list", async () => {
```
[the remark this answers is not in the document]
Comment (dominikbruechner) [reply src/components/organisms/HomeworksList.unit.js]: added test

**Review by martynakrysinska — APPROVED** [review]
Approved

**Pull request merged**